main.tf

```diff
@@ -1,8 +1,25 @@
-#################
-# Security group
-#################
+##########################
+# Security group with name
+##########################
 resource "aws_security_group" "this" {
-  count = "${var.create ? 1 : 0}"
+  count = "${var.create && var.use_name_prefix ? 0 : 1}"
+
+  name        = "${var.name}"
+  description = "${var.description}"
+  vpc_id      = "${var.vpc_id}"
+
+  tags = "${merge(var.tags, map("Name", format("%s", var.name)))}"
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+#################################
+# Security group with name_prefix
+#################################
+resource "aws_security_group" "this_name_prefix" {
+  count = "${var.create && var.use_name_prefix ? 1 : 0}"
```

Comment:
Don't switch values in the ternary operator, but rather use another condition.

`${var.create && (1 - var.use_name_prefix) ? 0 : 1}"`, for example.

Reply:
I changed the other one instead - makes more sense that way.